extensions/wikia/AdEngine/js/slot/resolveState.js

```diff
@@ -1,12 +1,17 @@
 /* global define */
 define('ext.wikia.adEngine.slot.resolveState', [
+	'ext.wikia.adEngine.context.uapContext',
+	'wikia.cache',
 	'wikia.log',
 	'wikia.querystring'
-], function (log, QueryString) {
+], function (uapContext, cache, log, QueryString) {
 	'use strict';
 
 	var logGroup = 'ext.wikia.adEngine.slot.resolveState',
-		qs = new QueryString();
+		qs = new QueryString(),
+		cacheKey = 'adEngine_resolvedStateCounter',
```

Comment:
How about setting cacheKey = `adEngine_resolvedStateCounter_<adID>`. Then you could get rid of forEach on record